PocketKit/Sources/PocketKit/Main/MainCoordinator.swift

```diff
@@ -9,6 +9,10 @@ class MainCoordinator {
     var viewController: UIViewController {
         regular.viewController
     }
+    
+    var tabBar: UITabBar? {
+        (compact.viewController as? UITabBarController)?.tabBar
```

Comment:
I don't love that we have to downcast to `UITabBarController` here. The fact that the compact view uses a tab bar controller is meant to be an implementation detail that other coordinators don't know/care about.

How about we go one level deeper and add this same property to `CompactMainCoordinator`?
```swift
class MainCoordinator {
    var tabBar: UITabBar {
        compact.tabBar
    }
}

class CompactMainCoordinator {
    // ...
    var tabBar: UITabBar {
        tabBarController.tabBar
    }
}
```

In fact, we could further hide implementation details of the compact coordinator if we changed this to simply return a view that we can anchor to when setting up constraints for the banner:

```swift
class MainCoordinator {
    var bannerAnchor: UIView {
        compact.bannerAnchor
    }
}

class CompactMainCoordinator {
    // ...
    var bannerAnchor: UIView {
        tabBarController.tabBar
    }
}
```

I'll leave this one up to you. I just see this downcast and a little warning buzzer goes off in my head but I can happily ignore it in this case if you can. 😄

Reply:
sounds good to me :) I went with the first one because we also need to use the tabBar to get the height, so not only using it for the anchor constraints